### PR TITLE
Add global identifier lookup

### LINKED ISSUE #62: Have find_symbol take an identifier as input

### PR DESCRIPTION
**Pull request by adamhutchings**
Closes #62, #63.

### CODE REVIEW

File: src/parser/parser.c [hunk 1]
```diff
@@ -176,7 +172,8 @@ int yfp_ident(struct yfcs_identifier * node, struct yf_lexer * lexer) {
             yfl_unlex(lexer, &tok);
             /* There's no prefix. */
             strcpy(node->name, node->filepath);
-            strcpy(node->filepath, lexer->input->identifier_prefix);
+            //strcpy(node->filepath, lexer->input->identifier_prefix);
```
Comment (JakeRoggenbuck) [inline src/parser/parser.c:175]: Keeping this in the code?

Comment (adamhutchings) [reply src/parser/parser.c:175]: Yeah, just in case we need it later.

Comment (JakeRoggenbuck) [reply src/parser/parser.c:175]: also, file tree beta looking nice
image

Comment (adamhutchings) [reply src/parser/parser.c:175]: Yo I didn't even know that existed!

Comment (JakeRoggenbuck) [reply src/parser/parser.c:175]: you gotta turn on the beta tho

**Review by JakeRoggenbuck — APPROVED** [review]

**Pull request merged**